runtime/vm/jfr.cpp

```diff
@@ -1016,11 +1024,11 @@ jfrThreadContextSwitchRate(J9VMThread *currentThread)
 
 			initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_THREAD_CONTEXT_SWITCH_RATE);
 
-			if (-1 == jfrState->prevContextSwitchTimestamp) {
-				jfrEvent->switchRate = 0;
+			if (-1 == jfrState->prevContextSwitchTimestamp || currentTime == jfrState->prevContextSwitchTimestamp) {
```

Comment:
If we keep the test for `-1` here, there's no need to use `j9time_nano_time()` at initialization time (but this is missing parentheses and should be split across multiple lines).

Reply:
My concern was if `omrsysinfo_get_number_context_switches` fails, `prevContextSwitchTimestamp` should fall back to `-1`.

Reply:
By checking for `-1` here, the value of calling `omrsysinfo_get_number_context_switches()` at initialization time is that it avoids having `jfrThreadContextSwitchRate()` report a zero context-switch rate the first time it is called. The initialization code is fine.

Please add the missing parentheses here and fix the formatting:
```c
			if ((-1 == jfrState->prevContextSwitchTimestamp)
				|| (currentTime == jfrState->prevContextSwitchTimestamp)
			) {
```

Reply:
Please also rebase to avoid the implicit merge with #20882.